opm/input/eclipse/EclipseState/Grid/GridDims.hpp

```diff
@@ -31,32 +30,30 @@ namespace Opm {
     class GridDims
     {
     public:
-
         GridDims();
-        explicit GridDims(std::array<int, 3> xyz);
-        GridDims(size_t nx, size_t ny, size_t nz);
+        explicit GridDims(const std::array<int, 3>& xyz);
+        GridDims(std::size_t nx, std::size_t ny, std::size_t nz);
 
         static GridDims serializeObject();
 
         explicit GridDims(const Deck& deck);
 
-        size_t getNX() const;
-
-        size_t getNY() const;
-        size_t getNZ() const;
-        size_t operator[](int dim) const;
+        std::size_t getNX() const;
+        std::size_t getNY() const;
+        std::size_t getNZ() const;
+        std::size_t operator[](int dim) const;
 
-        const std::array<int, 3> getNXYZ() const;
+        std::array<int, 3> getNXYZ() const;
```

Comment:
> Do you be any chance know, why we did we decided to return by value in the first place?

I don't know, but I can speculate.  The `GridDims` utility was written at a time when we did not have enough experience with move semantics and copying to know what things to do and what things to avoid and we've not revisited the API decisions since then.  I also suspect that we expected calls to `GridDims::getNXYZ()` to be rare.